Kwp simulation

ankayca opened the pull request:
KWP message frame added to simulation side. MC3360 and arduino uno used for tests. This my first pull request so I anything wrong my bad.

ankayca committed:
fix(OBD9141sim.h): initialize KWP simulator

ankayca committed:
fix(OBD9141sim.cpp): KWP simulation initiliaze added

ankayca committed:
fix(OBD9141sim.cpp): answeRequest func modified to send KWP frame

iwanders commented on src/OBD9141sim.h:
```diff
@@ -75,7 +75,7 @@ class OBD9141sim{
         void handleRequests();
         void answerRequest(uint8_t mode, uint8_t pid);
-
+        bool use_kwp_;
```
Please assign this to be false in the constructor, so next to 
https://github.com/iwanders/OBD9141/blob/838bd4a46490dfa8f70e25217fd4fff337e14fa3/src/OBD9141sim.cpp#L15

iwanders replied:
Please incorporate this feedback, other than that it looks good now, thanks for incorporating the others.

iwanders commented on src/OBD9141sim.cpp:
```diff
@@ -96,6 +102,11 @@ void OBD9141sim::answerRequest(uint8_t mode, uint8_t pid){
     message[3] = 0x40 + mode; // is this actually what happens!?
     message[4] = pid;
     message[5] = 0;
+    if(use_kwp_){
```
```suggestion
    if (use_kwp_){
```

iwanders commented on src/OBD9141sim.cpp:
```diff
@@ -96,6 +102,11 @@ void OBD9141sim::answerRequest(uint8_t mode, uint8_t pid){
     message[5] = 0;
+    if(use_kwp_){
+        // now we modify the message for KWP, the payload is the request_len - 3 header bytes
+        message[0] = 0x82 + (answer->len );
```
```suggestion
        message[0] = 0x82 + answer->len;
```

iwanders commented on src/OBD9141sim.cpp:
```diff
@@ -48,6 +48,12 @@ void OBD9141sim::initialize(){
 }
 
+void OBD9141sim::initializeKWP(){
+    use_kwp_ = true;
```
```suggestion
    this->use_kwp_ = true;
```

I know it's not necessary, but the other code follows this pattern, so lets keep it similar.

iwanders reviewed (changes requested):
Hey @ankayca, thanks for taking the time to file this PR. I'm always pleasantly surprised when the simulator is useful to anyone but me, as I only really wrote it as a development tool for my own projects.

I made a few minor remarks on the code, if you could incorporate those we should be able to merge it.

ankayca committed:
Update src/OBD9141sim.cpp

Co-authored-by: Ivor Wanders <iwanders@users.noreply.github.com>

ankayca committed:
Update src/OBD9141sim.cpp

Yes thats true actually I have no experience with C++ and class type so I just predict the simulatiries with C.

Co-authored-by: Ivor Wanders <iwanders@users.noreply.github.com>

ankayca committed:
Update src/OBD9141sim.cpp

Co-authored-by: Ivor Wanders <iwanders@users.noreply.github.com>

ankayca reviewed (commented):
I gonna continue to work with this simulator. Its very useful and there are not too much information about K line on the internet and thanks for your reviews.

iwanders commented:
One comment about adding the assignment of `use_kwp_` to false in the constructor wasn't addressed, please add that, other than that it looks good.

ankayca committed:
refactor(OBD9141sim.cpp): kwp boolean usage

kwp boolean assigned to false.

ankayca reviewed (commented):
my bad. assigned

iwanders commented:
Looks good now, I'm going to merge this.

Thanks for your contribution @ankayca !
